Sources/Client/Model/Channel/Channel+Requests.swift

```diff
@@ -133,6 +135,37 @@ public extension Channel {
         Client.shared.send(eventType: eventType, to: self, completion)
     }
     
+    /// Send a keystroke event for the current user.
+    /// - Parameter completion: a completion block with `Event`.
+    @discardableResult
+    func keystroke(_ completion: @escaping Client.Completion<Event>) -> Cancellable {
+        currentUserTypingTimerControl?.cancel()
+        
+        currentUserTypingTimerControl = DefaultTimer.schedule(timeInterval: 5, queue: .main) { [weak self] in
+            self?.stopTyping(completion)
+        }
+        
+        guard !isCurrentUserTyping else {
+            return Subscription.empty
+        }
+        
+        isCurrentUserTyping = true
```

Comment:
Maybe we should assign this after the request has succeeded? Or it can stay like this bu we should reset it if the request fails.